File: mineai-ui/src/api/mineai/importTransferTask.ts
import { maHttp } from '/@/utils/http/axios';
import { DubheBackendUrlEnum } from '/@/enums/mineaiEnum';

const prefix = DubheBackendUrlEnum.DUBHE_DATASET;
const base = 'import-tasks';

export interface ImportTransferTask {
  id: number;
  taskName: string;
  datasetType: 'IMAGE' | 'VIDEO' | 'POINT_CLOUD' | 'MULTI' | string;
  datasetId?: number;
  sourceType?: string;
  status: 'PENDING' | 'PROCESSING' | 'COMPLETED' | 'FAILED' | 'CANCELLED' | string;
  stage?: string;
  progress?: number;
  totalFiles?: number;
  successFiles?: number;
  failedFiles?: number;
  totalBytes?: number;
  transferredBytes?: number;
  retryCount?: number;
  errorMessage?: string;
  createTime?: string;
  updateTime?: string;
  events?: Array<{ eventType: string; message: string; detail?: string; createTime?: string }>;
}

export function getImportTransferTasks(params: Recordable = {}) {
  return maHttp.get({ url: base, params, headers: {} }, { urlPrefix: prefix });
}

export function getImportTransferTask(id: number) {
  return maHttp.get({ url: `${base}/${id}`, headers: {} }, { urlPrefix: prefix });
}

export function createImportTransferTask(data: Recordable) {
  return maHttp.post({ url: base, data, headers: {} }, { urlPrefix: prefix });
}

export function updateImportTransferProgress(id: number, data: Recordable) {
  return maHttp.post({ url: `${base}/${id}/progress`, data, headers: {} }, { urlPrefix: prefix });
}

export function completeImportTransferTask(id: number) {
  return maHttp.post({ url: `${base}/${id}/complete`, headers: {} }, { urlPrefix: prefix });
}

export function failImportTransferTask(id: number, message: string) {
  return maHttp.post(
    { url: `${base}/${id}/fail`, params: { message }, headers: {} },
    { urlPrefix: prefix },
  );
}

export function cancelImportTransferTask(id: number) {
  return maHttp.post({ url: `${base}/${id}/cancel`, headers: {} }, { urlPrefix: prefix });
}

export function retryImportTransferTask(id: number) {
  return maHttp.post({ url: `${base}/${id}/retry`, headers: {} }, { urlPrefix: prefix });
}

export function deleteImportTransferTasks(ids: number[]) {
  return maHttp.delete({ url: base, data: ids, headers: {} }, { urlPrefix: prefix });
}

File: dubhe-server/dubhe-data/src/main/java/org/dubhe/data/domain/entity/ImportTransferTask.java
package org.dubhe.data.domain.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.util.Date;

/** Persistent user-facing import task for the transfer center. */
@Data
@TableName("transfer_import_task")
public class ImportTransferTask {
    @TableId(type = IdType.AUTO)
    private Long id;
    private String taskName;
    private String datasetType;
    private Long datasetId;
    private String sourceType;
    private String status;
    private String stage;
    private Integer progress;
    private Integer totalFiles;
    private Integer successFiles;
    private Integer failedFiles;
    private Long totalBytes;
    private Long transferredBytes;
    private Integer retryCount;
    private String errorMessage;
    private Date startedAt;
    private Date finishedAt;
    private Date lastHeartbeatAt;
    private Long createUserId;
    private Date createTime;
    private Date updateTime;
    private Boolean deleted;
}
